design/open/oidc-support.md

```diff
@@ -0,0 +1,28 @@
+## OIDC support in lakeFS
+
+As suggested by @arielshaqed, we started researching the option to implement OIDC support as part of lakeFS.
+
+In such a solution, the person who configures lakeFS will supply the parameters required for implementing OIDC, including the provider's domain, the client ID and secret, and a callback URL. Once activated, the standard (non-OIDC) cookie based lakeFS authentication will not be available. API access will still use credentials managed internally like today.
+
+As a first step, authorization will still be managed internally by lakeFS.
+
+### Open questions
+
+#### What do we save in our database?
+What information about users should we save in the lakeFS database?
+
+The minimum here is to save the unique user ID generated by the OIDC provider (the "subject" claim).
+
+However, this may not be enough. 
+Since we only want authentication (and not authorization) to be handled by the external provider, we should be able to link between a user and roles/policies. It seems that to do so we need to save some more info (at least username/email). Once we also move authorizaiton outside, it may be easier to avoid saving any PII on the lakeFS side.
+
+#### Can an installation deactivate OIDC after it's been turned on?
+
+If so, how does one migrate the existing users?
+
+#### How does one create the initial admin user?
+
+#### How does the admin set the initial permissions for new users?
+
+The users will now be managed in an external database. Once they log in to lakeFS for the first time, a user will be created in the lakeFS database. How does the admin specify the permissions for such a user?
```

Comment:
how do new users today get assigned to default groups?
AFAIK it's not being handled right now, I think it's a good approach to allow a default-group (either via configuration or UI), but it doesn't change the current behavior.

Reply:
LDAP provides a default group.  Otherwise it would be pretty much unusable.

Reply:
> The difference is that today the admin can create a user and immediately set up permissions for them.
> 
> With my suggestion, the admin will invite the user, but the user in lakeFS will be created only once the user logs in, which can happen days later. Then the admin needs to remember to set the correct permissions.

I would stick with using a default group.
This way, on user-activation, it'll be assigned automatically, and if the admin would like to change it, he can, once the user is created.